### ISSUE TITLE
Unable to get access token for custom WebAPI for federated B2B users

### ISSUE THREAD

**Issue by biste5** [AzureAD/microsoft-authentication-library-for-js#1115]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Browser:
- [x] Chrome version 78
- [ ] Firefox version XX
- [ ] IE version XX
- [x] Edge version 80
- [ ] Safari version XX

## Library version

<pre><code>
Library version: 1.2.0-beta.3

</code></pre>

## Current behavior
We have a React SPA that uses Azure B2C for authenticating users. We can allow local B2C users and we have federated our B2B associated tenant so that also users included there can login via B2C. ([Documentation for federating a B2B tenant in B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/active-directory-b2c-setup-aad-custom))

We also have a WebAPI with its separate app registration and published scopes.
App registration for our SPA requests to be able to call into our WebAPI.
After we authenticate users, we then need to be able to acquire access tokens for users to call into our WebAPI.

B2C users can authenticate, and also get access token for WebAPI as expected.
B2B users are able to authenticate in the same way, but when trying to acquire access token for our WebAPI the operation fails ('_Token renewal operation failed due to timeout_')

## Expected behavior
The authentication and token acquisition flows should work consistently for B2B and B2C users on that B2C tenant.
Token acquisition for B2B users should succeed

## Minimal reproduction of the problem with instructions
- Configure a federation between a B2C and a B2B tenant, as per documentation linked above)
- Make sure there are two app registrations in the B2C tenant to represent the Web Client (SPA) and Web API (SPA's app reg must allow implicit flow)
- Configure the Web API app registration to expose a scope
- Configure the Web Client app registration to be able to call into Web API's published scope
- Perform authentication flow and token acquisition using MSAL for the WebAPI's published scope, using the App ID of Web Client using a B2C user, then a B2B user

**Comment by jmckennon** [comment]
@biste5 We recently pushed a bunch of fixes to various timeout-related bugs in `1.2.0-beta.4`, can you try upgrading and see if you're still seeing the same issues?

**Comment by biste5** [comment]
I have updated and performed some tests.
There's a difference in behavior, for which the redirects seems correctly followed, and not ending up in 'token renewal failed' error.

However, the flow in code of token acquisition doesn't work correctly yet.

I'll try to provide more insights and details about the precise behavior

**Comment by jmckennon** [comment]
@biste5 Do you have any updates here? Still seeing problems? I was waiting for more details, but it's been some time. 

We could take a look at the url that you're getting back and I can try to diagnose if there are still some problems.

We now also have `msal@1.2.0`, which I don't think will necessarily fix your issues, but it's always worth a shot.

**Comment by biste5** [comment]
Sorry but I didn't get chance to collect more details; also because we've dropped B2C for now and staying with B2B due of some issues.
To note/highlight anyway that for being able to assign the permission for the B2C app registration to our Web API resource (app registration), which lives in B2B, I had to manually edit the manifest for the B2C app registration to make it multi-tenant. Otherwise I can't see and assign permissions between the B2C (web client / SPA app registration) and the backend to have the first one acquiring token for the second one.

So I'm also not sure if I have gone into an unsupported scenario with this.
